tests/components/analytics_insights/conftest.py

```diff
@@ -7,10 +7,10 @@
 from python_homeassistant_analytics import CurrentAnalytics
 from python_homeassistant_analytics.models import CustomIntegration, Integration
 
-from homeassistant.components.analytics_insights import DOMAIN
 from homeassistant.components.analytics_insights.const import (
     CONF_TRACKED_CUSTOM_INTEGRATIONS,
     CONF_TRACKED_INTEGRATIONS,
+    DOMAIN,
```

Comment:
Ideally, outside code should not import from submodules (e.g. `const`). Everything needed should be exported in `__init__.py`.

_In this case `CONF_TRACKED_CUSTOM_INTEGRATIONS` is the only thing missing._

Reply:
We never really had this requirement. That would require integrations to provide a `__all__` block, do we really want that?

Reply:
> We never really had this requirement.

True. It's more a best practice. Basically consider everything within an integration private and only access the top-level from the outside. As long as that stays the same, you're free to move stuff inside around as you like. E.g. move `DOMAIN` from `__init__.py` to `const.py`.

> That would require integrations to provide a `__all__` block, do we really want that?

Not necessarily. An import with `# noqa: F401` is enough.

Reply:
I thought this rule only applied to components importing from other components, and didn't apply to component tests importing from their own component.

Reply:
> I thought this rule only applied to components importing from other components, and didn't apply to component tests importing from their own component.

It's not a defined rule in itself as far as I'm aware of. I'd just consider it good practice.
However, that doesn't need to hold up this PR thus I approved it just now.